pkg/scheduler/core/division_algorithm.go

```diff
@@ -0,0 +1,205 @@
+	preference policyv1alpha1.ReplicaDivisionPreference, preUsedClustersName ...string) ([]workv1alpha2.TargetCluster, error) {
+	switch preference {
+	case policyv1alpha1.ReplicaDivisionPreferenceAggregated:
+		return divideReplicasByAggregation(clusterAvailableReplicas, replicas, preUsedClustersName...), nil
+	case policyv1alpha1.ReplicaDivisionPreferenceWeighted:
+		return divideReplicasByAvailableReplica(clusterAvailableReplicas, replicas, clustersMaxReplicas), nil
+	default:
+		return nil, fmt.Errorf("undefined replicaSchedulingType： %v", preference)
+	}
+}
+
+func divideReplicasByAggregation(clusterAvailableReplicas []workv1alpha2.TargetCluster,
+	replicas int32, preUsedClustersName ...string) []workv1alpha2.TargetCluster {
+	clusterAvailableReplicas = presortClusterList(clusterAvailableReplicas, preUsedClustersName...)
+	clustersNum, clustersMaxReplicas := 0, int32(0)
+	for _, clusterInfo := range clusterAvailableReplicas {
+		clustersNum++
+		clustersMaxReplicas += clusterInfo.Replicas
+		if clustersMaxReplicas >= replicas {
+			break
+		}
+	}
+	var unusedClusters []string
+	for i := clustersNum; i < len(clusterAvailableReplicas); i++ {
+		unusedClusters = append(unusedClusters, clusterAvailableReplicas[i].Name)
+	}
+	return divideReplicasByAvailableReplica(clusterAvailableReplicas[0:clustersNum], replicas, clustersMaxReplicas, unusedClusters...)
+}
+
+func divideReplicasByAvailableReplica(clusterAvailableReplicas []workv1alpha2.TargetCluster, replicas int32,
+	clustersMaxReplicas int32, unusedClusters ...string) []workv1alpha2.TargetCluster {
+	desireReplicaInfos := make(map[string]int32)
+	allocatedReplicas := int32(0)
+	for _, clusterInfo := range clusterAvailableReplicas {
+		desireReplicaInfos[clusterInfo.Name] = clusterInfo.Replicas * replicas / clustersMaxReplicas
+		allocatedReplicas += desireReplicaInfos[clusterInfo.Name]
+	}
+
+	if remainReplicas := replicas - allocatedReplicas; remainReplicas > 0 {
+		for i := 0; remainReplicas > 0; i++ {
+			desireReplicaInfos[clusterAvailableReplicas[i].Name]++
+			remainReplicas--
+			if i == len(desireReplicaInfos) {
+				i = 0
+			}
+		}
+	}
+
+	// For scaling up
+	for _, cluster := range unusedClusters {
+		if _, exist := desireReplicaInfos[cluster]; !exist {
+			desireReplicaInfos[cluster] = 0
+		}
+	}
+
+	targetClusters := make([]workv1alpha2.TargetCluster, len(desireReplicaInfos))
+	i := 0
+	for key, value := range desireReplicaInfos {
+		targetClusters[i] = workv1alpha2.TargetCluster{Name: key, Replicas: value}
+		i++
+	}
+	return targetClusters
+}
+
+func scaleScheduleByReplicaDivisionPreference(spec *workv1alpha2.ResourceBindingSpec, preference policyv1alpha1.ReplicaDivisionPreference,
+	preSelectedClusters []*clusterv1alpha1.Cluster) ([]workv1alpha2.TargetCluster, error) {
+	assignedReplicas := util.GetSumOfReplicas(spec.Clusters)
+	if assignedReplicas > spec.Replicas {
+		newTargetClusters, err := scaleDownScheduleByReplicaDivisionPreference(spec, preference)
```

Comment:
Now only support `Replicas` scales, right? Not support `request` spec changes.

Reply:
That's right. We could add a new ScheduleType `ExpandResourceSchedule`. @qianjun1993 has implemented this type in our inner version. It could be contributed to `karmada` soon later.

Reply:
> That's right. We could add a new ScheduleType `ExpandResourceSchedule`. @qianjun1993 has implemented this type in our inner version. It could be contributed to `karmada` soon later.

Hi, I don't understand why we should add a new ScheduleType? I'd prefer to consider it as part of reschedule.

Reply:
It seems not like a `ScaleSchedule` type. When the resource request reduces, we don't have to reschedule the replicas because the pod could be recreated successfully. But when the resource request expands, we should reschedule all replicas, it is more like a `FirstSchedule` instead of `ScaleSchedule`. All the pods will be destroyed and the new ones will be created by `deployment_controller`. These replicas need to reschedule again.

So I think the new type may share the same logic with `FirstSchedule`. But the judgement of the type will be different from any other types.

Reply:
> Hi, I don't understand why we should add a new ScheduleType? I'd prefer to consider it as part of reschedule.

That's right. I think we have the same opinion. But when reducing, we don't have to reschedule. Just let the result be the same as before. The new type is just for record, not for another logic. The relationship is like `FirstSchedule` and `ReconcileSchedule`. They are different types but share the same scheduling logic.

Reply:
> It seems not like a `ScaleSchedule` type. When the resource request reduces, we don't have to reschedule the replicas because the pod could be recreated successfully. But when the resource request expands, we should reschedule all replicas, it is more like a `FirstSchedule` instead of `ScaleSchedule`. All the pods will be destroyed and the new ones will be created by `deployment_controller`. These replicas need to reschedule again.
> 
> So I think the new type may share the same logic with `FirstSchedule`. But the judgement of the type will be different from any other types.

When the resource request expands, the key point is the old pods will take up resources which will effect the schedule accuracy.

Reply:
> e new type is just for re

Got it. You don't mean to add a new type in API, right?

Reply:
You are right. I can't figure out a better way. Only if the estimator could ignore these pods. It may be a little complicated.

Reply:
> Got it. You don't mean to add a new type in API, right?

Yes. I don't mean that. Just a code type.

Reply:
> That's right. We could add a new ScheduleType ExpandResourceSchedule. @qianjun1993 has implemented this type in our inner version. It could be contributed to karmada soon later.

@Garrybest @qianjun1993 
Hold on. On the contrary, I'm thinking reduce the `ScheduleType`, too much schedule type now and hard to maintain.
The idea would be all schedule types share the same scheduling process.

Reply:
Good idea. It makes sense. We may need more discussion and adaption.

Reply:
The ExpandResourceSchedule like FirstSchedule and ReconcileSchedule in our inner version is implemented when I add the assignReplicas step to scheduler .  It shares the same scheduling process with FirstSchedule and ReconcileSchedule . But now I also find it has some problems that both ExpandResourceSchedule and ReconcileSchedule may different from FirstSchedule in #900.